src/docfx/lib/JsonUtility.cs

```diff
@@ -157,6 +157,17 @@ public static (List<Error>, T) Deserialize<T>(string json)
             return (errors, result);
         }
 
+        /// <summary>
+        /// De-serialize a text reader to an object
+        /// </summary>
+        public static (List<Error>, T) Deserialize<T>(TextReader reader)
```

Comment:
use the `string` version for now. we'll need a perf test to see the tradeoff between `TextReader` over `string` later, and apply it across the board